### ISSUE TITLE
Wrong dimension for the first layer visualization

### ISSUE THREAD

**Issue by zhangyu2ustc** [utkuozbulak/pytorch-cnn-visualizations#52]
Hi,
  I was using the guided backpropagation to visualize my own pretrained models and found out the resultant gradient maps did not have the same dimension as the input image, but the ouput shape of the first layer.
 For instance, I have the input of [batch, 1000, 17] and the ouput of the first layer is [batch,1000,32]
Then, the gradient maps has the shape of [1000,32] with the elements from 17~32 as non zeros.

I have also tried to print the shape of the gradient in the hook function, which is also [batch, 1000, 32]
Can you help me to figure out what went wrong?

Thank you very much.
Best,
   Yu.

**Comment by utkuozbulak** [comment]
The example code here does not use any batch. You should modify the code to enable batches.

**Comment by maryjis** [comment]
Hi, I have a similar problem! I use my own pretrained models ( pretrained vgg16)  and get gradient maps did not have the same dimension as the input image.
For example, I use  torch.Size([batch, 3, 224, 224]), but in hook function I get 
            print(grad_in[0]): None
            print(grad_in[1].shape):torch.Size([64, 3, 3, 3]
            print(grad_in[2].shape): torch.Size([64])
